internal/fs/inode/file.go

```diff
@@ -1049,3 +1053,17 @@ func (f *FileInode) InitBufferedWriteHandlerIfEligible(ctx context.Context) (boo
 	}
 	return false, nil
 }
+
+func (f *FileInode) areBufferedWritesSupported(openMode util.OpenMode, obj *gcs.Object) bool {
+	// For new files and existing files of size 0, buffered writes are always supported.
+	if f.local || obj.Size == 0 {
+		return true
+	}
+
+	if f.config.Write.ExperimentalEnableRapidAppends {
```

Comment:
we should always do short-circuits instead of having nested if-else

if !f.config.write.experimentalEnableRapidAppends {
   return false.
}